optimizer/lib/transformers/AddBlurryImagePlaceholders.js

```diff
@@ -155,6 +173,49 @@ class AddBlurryImagePlaceholders {
     });
     return false;
   }
+
+  /**
+   * Checks if an image should have a blurred image placeholder.
+   * The current criteria for determining if a blurry image placeholder should
+   * be appended is as follows:
+   * - The source for the image should be a JPEG.
+   * - If the element is an amp-img that is responsive and does not have a no
+   * loading attribute OR the element is a poster on an amp-video
+   *
+   * This criteria was found to be the most common places where a blurry image
+   * placeholder would likely want to be used through manual examination of
+   * existing AMP pages.
+   * @param {Node} node The DOM element that is being checked to see if it
+   * should have a blurred placeholder.
+   * @param {string} src The image source that is being checked.
+   * @param {string} tagName The type of element that is being checked.
+   * @return {boolean} Whether or not the element should have a blurred
+   * placeholder child.
+   * @private
+   */
+  shouldAddBlurryPlaceholder_(node, src, tagName) {
+    // Checks to see if the image doesn't have a source or if it already has a
+    // placeholder.
```

Comment:
More useful comments describe the "why" rather than the "what", unless the "what" is really complex.